### PR TITLE
Add conv2d and max_pool2d functions

### PR DESCRIPTION
**Pull request by vbkaisetsu**
This branch adds `conv2d` and `max_pool2d` functions, and updates test cases.

### CODE REVIEW

File: primitiv/_function.pyx [hunk 1]
```diff
@@ -205,6 +205,20 @@ class functions:
     def stop_gradient(Node x):
         return wrapNode(func_stop_gradient(x.wrapped))
 
+    @staticmethod
+    def conv2d(Node x, Node w, unsigned padding0, unsigned padding1, unsigned stride0, unsigned stride1, unsigned dilation0, unsigned dilation1):
```
Comment (odashi) [inline primitiv/_function.pyx]: These lines look too long to show and I recommend to add some newlines appropriately similar to the next line.

Comment (vbkaisetsu) [reply primitiv/_function.pyx]: Fixed.

**Review by odashi — DISMISSED** [review]

**Comment by odashi** [comment]
Let's add XavierNormalConv2D as well.

**Comment by vbkaisetsu** [comment]
Added.

**Review by odashi — DISMISSED** [review]

**Review by odashi — APPROVED** [review]

**Pull request merged**